src/cares_wrap.cc

```diff
@@ -122,10 +122,22 @@ struct node_ares_task {
   ChannelWrap* channel;
   ares_socket_t sock;
   uv_poll_t poll_watcher;
-  RB_ENTRY(node_ares_task) node;
 };
 
-RB_HEAD(node_ares_task_list, node_ares_task);
+struct TaskHash {
+  size_t operator()(node_ares_task* a) const {
+    return std::hash<node_ares_task*>()(a);
+  }
+};
+
+struct TaskEqual {
+  inline bool operator()(node_ares_task* a, node_ares_task* b) const {
+    return a->sock == b->sock;
+  }
+};
+
+typedef std::unordered_set<node_ares_task*, TaskHash, TaskEqual>
```

Comment:
Ah, sorry, I missed that I hadn’t pushed the right code for `TaskHash` yet. Now it’s more obvious why that doesn’t work. :)